tutorials/introductory/animation_tutorial.py

```diff
@@ -0,0 +1,249 @@
+    # Therefore, we create the entire data in each frame to draw
+    x = np.sin(a * t[:frame] + delta)
+    y = np.sin(b * t[:frame])
+    data = np.stack([x, y]).T
+    scat.set_offsets(data)
+    return (scat,)
+
+
+ani = animation.FuncAnimation(fig=fig, func=update, frames=400, interval=30)
+plt.show()
+
+
+###############################################################################
+# Animating :class:`~matplotlib.image.AxesImage`
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# When we plot an image using `.pyplot.imshow`, it returns an
+# :class:`~matplotlib.image.AxesImage` object. The data in this object can also
+# similarly be modified by using the `.image.AxesImage.set_data` method.
+
+fig, ax = plt.subplots()
+rng = np.random.default_rng()
+
+aximg = ax.imshow(rng.uniform(low=0, high=1, size=(10, 10)), cmap="Blues")
+
+
+def update(frame):
+    data = rng.uniform(low=0, high=1, size=(10, 10))
+    aximg.set_data(data)
+    return (aximg,)
+
+
+ani = animation.FuncAnimation(fig=fig, func=update, frames=None, interval=200)
+plt.show()
+
+###############################################################################
+# :class:`~matplotlib.animation.ArtistAnimation`
+# ----------------------------------------------
+#
+# On the other hand, :class:`~matplotlib.animation.ArtistAnimation` can be used
+# to generate animations if we have data on various different artists. This
+# list of artists is then converted frame by frame into an animation.
+
+
+fig, ax = plt.subplots()
+ax.grid()
+rng = np.random.default_rng()
+
+x_frames = rng.uniform(low=0, high=1, size=(100, 120))
+y_frames = rng.uniform(low=0, high=1, size=(100, 120))
+artists = [
+    [ax.scatter(x_frames[:, i], y_frames[:, i], c="b")]
+    for i in range(x_frames.shape[-1])
+]
+
+ani = animation.ArtistAnimation(fig=fig, artists=artists, repeat_delay=1000)
+plt.show()
+
+###############################################################################
+# Animation Writers
+# =================
+#
+# Animation objects can be saved to disk using various multimedia writers
+# (ex: Pillow, *ffpmeg*, *imagemagick*). Not all video formats are supported
+# by all writers. There are 4 major types of writers:
+#
+# - :class:`~matplotlib.animation.PillowWriter` - Uses the Pillow library to
+#   create the animation.
+#
+# - :class:`~matplotlib.animation.HTMLWriter` - Used to create JS-based
+#   animations.
+#
+# - Pipe-based writers - :class:`~matplotlib.animation.FFMpegWriter` and
+#   :class:`~matplotlib.animation.ImageMagickWriter` are pipe based writers.
+#   These writers pipe each frame to the utility (*ffmpeg* / *imagemagick*)
+#   which then stitches all of them together to create the animation.
+#
+# - File-based writers - :class:`~matplotlib.animation.FFMpegFileWriter` and
+#   :class:`~matplotlib.animation.ImageMagickFileWriter` are examples of
+#   file-based writers. These writers are slower than their standard writers
```

Comment:
`FFMpegWriter` for `FFMpegFileWriter` and similarly for Imagemagick but I guess calling them non-file based writers instead of standard would be better, yes.